web3/main.py

```diff
@@ -132,11 +138,12 @@ def providers(self, providers):
         self.manager.providers = providers
 
     @staticmethod
+    @deprecated_for("This method has been renamed to web3.keccak")
     @apply_to_return_value(HexBytes)
     def sha3(primitive=None, text=None, hexstr=None):
```

Comment:
I'm worried that in the future we're going to change something in one implementation or the other, instead of both. Can we just make the deprecated version an alias to the new one, like:

```py
def sha3(...):
   return Web3.keccak(...)
```

I think it's okay to have the error message reference keccak instead of sha3 during deprecation time.